docs/source/testing.rst

```diff
@@ -785,6 +773,31 @@ Some decorators like ``@parameterized`` rewrite test names, therefore ``@slow``
     @slow
     def test_integration_foo():
 
+As explained at the beginning of this document, slow tests get to run on a scheduled basis, rather than in PRs CI checks. So it's possible that some problems will be missed during a PR submission and get merged. Such problems will get caught during the next scheduled CI job. But it also means that it's important to run the slow tests on your machine before submitting the PR.
+
+Here is a rough decision making mechanism for choosing which tests should be marked as slow:
+
+If the test is focused on one of the library's internal components (e.g., modeling files, tokenization files, pipelines), then we should run that test in the non-slow test suite. If it's focused on an other aspect of the library, such as the documentation or the examples, then we should run these tests in the slow test suite. And then, to refine this approach we should have exceptions:
+
+* All tests that need a specific set of weights (e.g., model or tokenizer integration tests, pipeline integration tests) should be set to slow.
+* All tests that need to do a training (e.g, trainer integration tests) should be set to slow.
```

Comment:
Not sure how to rephrase this one as it's untrue: we have some training on small datasets (for the regression problem in the trainer tests, or to overfit on a mini-batch in the test-examples script). Maybe:
```suggestion
* All tests that need to do a training not specifically optimized to be fast should be set to slow.
```

Reply:
(maybe out of scope)

I also think regardless of how tests are marked, authors should attempt to make their tests faster. Try training fewer epochs, try reducing `seq_len` etc, use fp16 if available, etc. 

Maybe this doc needs a separate section called

Tests that involve training



I also have fast examples/ tests that are not marked slow. Like @sgugger 's, they took a lot of work and this type of work should not be discouraged.

Reply:
This is a fantastic idea, @sshleifer! Would you like to write a stab of it and we can work on it in another PR?